### ISSUE TITLE
CMakeList.txt fix deprecation warning from CMake.

### ISSUE THREAD

**Issue by alecazam** [zeux/meshoptimizer#803]
MeshOptimizer is specifying a min cmake this is deprecated.  So this generates unneeded warning messages.  Seems like a simple fix.  Just need a version number higher than 3.5 to suppress this.

change:
cmake_minimum_required(VERSION 3.0)

to:
cmake_minimum_required(VERSION 3.0...3.20)

https://cmake.org/cmake/help/latest/command/cmake_policy.html#version

Changed in version 3.31: Compatibility with versions of CMake older than 3.10 is deprecated. Calls to cmake_minimum_required(VERSION) or cmake_policy(VERSION) that do not specify at least 3.10 as their policy version (optionally via ...<max>) will produce a deprecation warning in CMake 3.31 and above.

```
CMake Deprecation Warning at .../meshoptimizer-src/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.
```

**Comment by zeux** [comment]
The minimum version has been 3.5 since July 2023, for three consecutive releases (0.20, 0.21, 0.22).

**Comment by zeux** [comment]
Ah, I missed the 3.31 note.

**Comment by zeux** [comment]
Fixed by a64a212836d471756f86121aa5afc3547ff6196c
